src/sage/matroids/oriented_matroids/circuit_oriented_matroid.py

```diff
@@ -0,0 +1,228 @@
+
+This implements an oriented matroid using the circuit axioms.
+
+AUTHORS:
+
+- Aram Dermenjian (2019-07-12): Initial version
+"""
+
+##############################################################################
+#       Copyright (C) 2019 Aram Dermenjian <aram.dermenjian.math at gmail.com>
+#
+#  Distributed under the terms of the GNU General Public License (GPL)
+#
+#  The full text of the GPL is available at:
+#
+#                  http://www.gnu.org/licenses/
+##############################################################################
+
+from sage.matroids.oriented_matroids.oriented_matroid import OrientedMatroid
+from sage.categories.sets_cat import Sets
+
+
+class CircuitOrientedMatroid(OrientedMatroid):
+    r"""
+    An oriented matroid implemented using circuit axioms.
+
+    This implements an oriented matroid using the circuit axioms. For this
+    let `\mathcal{C}` be a set of signed subsets and `E` a groundset. Then
+    a pair `M = (E, \mathcal{C})` is an oriented matroid using the circuit
+    axioms if (see Definition 3.2.1 in [BLSWZ1999]_):
+
+    - `\emptyset \notin \mathcal{C}`
+    - `\mathcal{C} = -\mathcal{C}`
+    - For all `X, Y \in \mathcal{C}`, if the support of `X` is contained
+      in the support of `Y` then `X = Y` or `X = -Y`
+    - For all `X, Y \in \mathcal{C}`, `X \neq -Y`, and
+      `e \in X^+ \cap Y^-` there exists a `Z \in \mathcal{C}` such that
+      `Z^+ \subseteq (X^+ \cup Y^+) \backslash \left\{e\right\}` and
+      `Z^- \subseteq (X^- \cup Y^-) \backslash \left\{e\right\}`.
+
+    INPUT:
+
+    - ``data`` -- a tuple containing :class:`SignedSubsetElement` elements or
+      data that can be used to construct :class:`SignedSubsetElement` elements
+    - ``groundset`` -- (default: ``None``) the groundset for the data; if not
+      provided, we grab the data from the signed subsets
+
+    EXAMPLES::
+
+        sage: from sage.matroids.oriented_matroids.oriented_matroid import OrientedMatroid
+        sage: M = OrientedMatroid([[1], [-1]], key='circuit'); M
+        Circuit oriented matroid of rank 0
+        sage: M.groundset()
+        (0,)
+
+        sage: C = [((1,4),(2,3)), ((2,3),(1,4))]
+        sage: M = OrientedMatroid(C, key='circuit'); M
+        Circuit oriented matroid of rank 3
+        sage: M.groundset()
+        (1, 2, 3, 4)
+        sage: M.an_element() in M.elements()
+        True
+        sage: M.elements() == M.circuits()
+        True
+
+        sage: C5 = [((1,),(3,),(2,)), ((1,2),(3,),(4,)),
+        ....:       ((3,),(1,),(2,)), ((3,),(1,2),(4,))]
+        sage: OrientedMatroid(C5, key='circuit')
+        Traceback (most recent call last):
+        ...
+        ValueError: groundsets must be the same
+
+    .. SEEALSO::
+
+        - :class:`~oriented_matroids.oriented_matroid.OrientedMatroid`
+        - :class:`~oriented_matroids.oriented_matroids_category.OrientedMatroids`
+    """
+
+    def __init__(self, data, groundset=None, category=None):
+        """
```

Comment:
```suggestion
    .. SEEALSO::

        - :class:`~sage.oriented_matroids.oriented_matroid.OrientedMatroid`
        - :class:`~sage.oriented_matroids.oriented_matroids_category.OrientedMatroids`
    """
    def __init__(self, data, groundset=None, category=None):
        """
```
Plus the `__init__` needs a doctest.